security-framework/src/secure_transport.rs

```diff
@@ -673,7 +787,7 @@ impl SslContext {
         unsafe {
             let ret = SSLSetIOFuncs(self.0, read_func::<S>, write_func::<S>);
             if ret != errSecSuccess {
-                return Err(HandshakeError::Failure(Error::new(ret)));
+                return Err(HandshakeError::Failure(ErrorCode::from_i32(ret).unwrap()));
```

Comment:
`unwrap` isn't great, but the only other alternative I can think of is to create a catch-all `UnknownError(i32)` variant on the `ErrorCode` enum.